File: app/src/main/java/com/calyptus/thikline/drawingcomponents/DrawingTool.java
package com.calyptus.thikline.drawingcomponents;

import android.graphics.Canvas;
import android.view.MotionEvent;
import android.view.View;

public interface DrawingTool {
    void setTouchEvent(View v, MotionEvent event);
    void setView(View drawingView);
    void onDraw(Canvas canvas);
}

File: app/src/main/java/com/calyptus/thikline/drawingcomponents/DrawingTool.kt
package com.calyptus.thikline.drawingcomponents

import android.graphics.Canvas
import android.view.MotionEvent
import android.view.View

interface DrawingTool {
    fun setTouchEvent(v: View?, event: MotionEvent?)
    fun setView(drawingView: View?)
    fun setDrawings(drawingData: DrawingData?)
}